vald/evm/evm.go

```diff
@@ -447,12 +423,16 @@ func (mgr Mgr) isTxReceiptFinalized(chain nexus.ChainName, txReceipt *geth.Recei
 	}
 
 	latestFinalizedBlockNumber, err := client.LatestFinalizedBlockNumber(context.Background(), confHeight)
-	if err != nil || latestFinalizedBlockNumber.Cmp(txReceipt.BlockNumber) < 0 {
+	if err != nil {
 		return false, err
 	}
 
 	mgr.latestFinalizedBlockCache.Set(chain, latestFinalizedBlockNumber)
 
+	if latestFinalizedBlockNumber.Cmp(txReceipt.BlockNumber) < 0 {
+		return false, err
```

Comment:
```suggestion
		return false, nil
```
I know it's the same, but you don't need to play with the logic anymore since it's not in a separate if-block.